src/framework/components/element/pointer-events.js

```diff
@@ -0,0 +1,190 @@
+pc.extend(pc, function() {
+
+    var PointEventsManager = {
```

Comment:
Do challenge this a lot, if we actually do want to provide API to deal with input, or API to deal with "picking" on elements. As input sounds to me as very specific to application, but "picking" on elements, can be used in many cases, not only mouse/touch inputs, but VR as well, and many others where not actual input is used.

Plus making touch/mouse API this way, is it preserving all the flexibility, or is somewhat restrictive to some use cases. Then I would really consider going "picking" approach instead of implementing whole mouse/touch handling.